p2p/src/test/kotlin/maru/p2p/P2PTest.kt

```diff
@@ -0,0 +1,387 @@
+    val privKey = unmarshalPrivateKey(privateKeyBytes.toArrayUnsafe())
+    val peerId = PeerId.fromPubKey(privKey.publicKey()).toBase58()
+
+    println("peerId: $peerId")
+    println("private key: $toHexString")
+
+    val p2pManager1 = P2PManager(key1, P2P(IPV4, PORT1, emptyList()))
+    val p2pManager2 = P2PManager(key2, P2P(IPV4, PORT2, emptyList()))
+    try {
+      p2pManager1.start()
+      val p2pNetwork1 = p2pManager1.p2pNetwork
+
+      p2pManager2.start()
+      val p2pNetwork2 = p2pManager2.p2pNetwork
+
+      p2pManager1.addStaticPeer(MultiaddrPeerAddress.fromAddress(PEER_ADDRESS_NODE_2))
+
+      Awaitility
+        .await()
+        .timeout(
+          6000,
+          TimeUnit.MILLISECONDS,
+        ).untilAsserted { assertThat(p2pNetwork1.peerCount).isEqualTo(1) }
+      Awaitility
+        .await()
+        .timeout(
+          6000,
+          TimeUnit.MILLISECONDS,
+        ).untilAsserted { assertThat(p2pNetwork2.peerCount).isEqualTo(1) }
+    } finally {
+      p2pManager1.stop()
+      p2pManager2.stop()
+    }
+  }
+
+  @Test
+  fun `static peers can be removed`() {
+    val p2pManager1 = P2PManager(key1, P2P(IPV4, PORT1, emptyList()))
+    val p2pManager2 = P2PManager(key2, P2P(IPV4, PORT2, emptyList()))
+    try {
+      p2pManager1.start()
+      val p2pNetwork1 = p2pManager1.p2pNetwork
+
+      p2pManager2.start()
+      val p2pNetwork2 = p2pManager2.p2pNetwork
+
+      p2pManager1.addStaticPeer(MultiaddrPeerAddress.fromAddress(PEER_ADDRESS_NODE_2))
+
+      Awaitility
+        .await()
+        .timeout(
+          6000,
+          TimeUnit.MILLISECONDS,
+        ).untilAsserted { assertThat(p2pNetwork1.peerCount).isEqualTo(1) }
+      Awaitility
+        .await()
+        .timeout(
+          6000,
+          TimeUnit.MILLISECONDS,
+        ).untilAsserted { assertThat(p2pNetwork2.peerCount).isEqualTo(1) }
+
+      p2pManager1.removeStaticPeer(MultiaddrPeerAddress.fromAddress(PEER_ADDRESS_NODE_2))
+
+      Awaitility
+        .await()
+        .timeout(
+          6000,
+          TimeUnit.MILLISECONDS,
+        ).untilAsserted { assertThat(p2pNetwork1.peerCount).isEqualTo(0) }
+      Awaitility
+        .await()
+        .timeout(
+          6000,
+          TimeUnit.MILLISECONDS,
+        ).untilAsserted { assertThat(p2pNetwork2.peerCount).isEqualTo(0) }
+    } finally {
+      p2pManager1.stop()
+      p2pManager2.stop()
+    }
+  }
+
+  @Test
+  fun `static peers can be configured`() {
+    val p2pManager1 = P2PManager(key1, P2P(IPV4, PORT1, emptyList()))
+    val p2pManager2 = P2PManager(key2, P2P(IPV4, PORT2, listOf(PEER_ADDRESS_NODE_1)))
+    try {
+      p2pManager1.start()
+      val p2pNetwork1 = p2pManager1.p2pNetwork
+
+      p2pManager2.start()
+      val p2pNetwork2 = p2pManager2.p2pNetwork
+
+      Awaitility
+        .await()
+        .timeout(
+          6000,
+          TimeUnit.MILLISECONDS,
+        ).untilAsserted { assertThat(p2pNetwork1.peerCount).isEqualTo(1) }
```

Comment:
Consider introducing a function `networkHasPeers(network: LibP2PNetwork, peers: Int)` so the tests are easier to read

Reply:
done